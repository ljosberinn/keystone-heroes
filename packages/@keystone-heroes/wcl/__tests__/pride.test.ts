import { graphql } from "msw";
import { setupServer } from "msw/node";

import {
  getManifestationOfPrideSourceID,
  getManifestationOfPrideDeathEvents,
  getDamageDoneToManifestationOfPrideEvents,
  getDamageTakenByManifestationOfPrideEvents,
} from "../src/queries/events/affixes/prideful";
import manifestationOfPrideAllDamageDoneEvents from "./fixtures/manifestationOfPrideAllDamageDoneEvents.json";
import manifestationOfPrideAllDamageTakenEvents from "./fixtures/manifestationOfPrideAllDamageTakenEvents.json";
import manifestationOfPrideDeathEvents from "./fixtures/manifestationOfPrideDeathEvents.json";
import spitefulPridefulEnemyNPCIDs from "./fixtures/spitefulPridefulEnemyNPCIDs.json";

describe("prideful", () => {
  const server = setupServer();

  beforeAll(() => {
    server.listen();
  });

  afterEach(() => {
    server.resetHandlers();
  });

  afterAll(() => {
    server.close();
  });

  test("getManifestationOfPrideSourceID", async () => {
    server.use(
      graphql.query("EnemyNPCIds", (_req, res, ctx) => {
        return res(ctx.data(spitefulPridefulEnemyNPCIDs));
      })
    );

    const result = await getManifestationOfPrideSourceID({
      reportID: "",
      fightID: 1,
    });

    expect(result).toMatchSnapshot();
  });

  test("getManifestationOfPrideDeathEvents", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: manifestationOfPrideDeathEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getManifestationOfPrideDeathEvents({
      reportID: "",
      fightID: 1,
      sourceID: 33,
      startTime: 0,
      endTime: 1,
    });

    expect(result).toMatchSnapshot();
  });

  // see https://github.com/ljosberinn/keystone-heroes/issues/31
  // technically the snapshot is invalid:
  // 8 entries instead of 5 due to 3 pets being active
  test("getDamageDoneToManifestationOfPrideEvents (all events)", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: manifestationOfPrideAllDamageDoneEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getDamageDoneToManifestationOfPrideEvents({
      reportID: "",
      startTime: 0,
      endTime: 1,
      targetID: 1,
    });

    expect(result).toMatchSnapshot();
  });

  test("getDamageDoneToManifestationOfPrideEvents (first event only)", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: manifestationOfPrideAllDamageDoneEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getDamageDoneToManifestationOfPrideEvents(
      {
        reportID: "",
        startTime: 0,
        endTime: 1,
        targetID: 1,
      },
      true
    );

    expect(result).toMatchSnapshot();
  });

  test("getDamageTakenByManifestationOfPrideEvents (all events)", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: manifestationOfPrideAllDamageTakenEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getDamageTakenByManifestationOfPrideEvents({
      reportID: "",
      startTime: 0,
      endTime: 1,
      targetID: 1,
    });

    expect(result).toMatchSnapshot();
  });

  test("getDamageTakenByManifestationOfPrideEvents (first event only)", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: manifestationOfPrideAllDamageTakenEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getDamageTakenByManifestationOfPrideEvents(
      {
        reportID: "",
        startTime: 0,
        endTime: 1,
        targetID: 1,
      },
      true
    );

    expect(result).toMatchSnapshot();
  });
});
