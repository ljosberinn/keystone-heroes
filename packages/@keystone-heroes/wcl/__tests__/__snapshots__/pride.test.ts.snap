// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prideful getDamageDoneToManifestationOfPrideEvents (all events) 1`] = `
Array [
  Object {
    "abilityGameID": 225919,
    "amount": 50444,
    "buffs": "337547.",
    "hitType": 1,
    "mitigated": 263,
    "overheal": undefined,
    "sourceID": 1,
    "sourceMarker": 7,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 196204,
    "type": "damage",
    "unmitigatedAmount": 879,
  },
  Object {
    "abilityGameID": 164812,
    "amount": 65468,
    "buffs": "308488.24858.307185.342814.",
    "hitType": 1,
    "overheal": undefined,
    "sourceID": 7,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 197371,
    "type": "damage",
    "unmitigatedAmount": 392,
  },
  Object {
    "abilityGameID": 213243,
    "amount": 58661,
    "buffs": "331939.324748.307185.208628.",
    "hitType": 2,
    "overheal": undefined,
    "sourceID": 6,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 198180,
    "type": "damage",
    "unmitigatedAmount": 1723,
  },
  Object {
    "abilityGameID": 345495,
    "amount": 3257,
    "buffs": "308488.24858.307185.342814.",
    "hitType": 2,
    "overheal": undefined,
    "sourceID": 16,
    "sourceInstance": 7,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 199711,
    "type": "damage",
    "unmitigatedAmount": 513,
  },
  Object {
    "abilityGameID": 1,
    "amount": 1643,
    "buffs": "330380.",
    "hitType": 1,
    "mitigated": 48,
    "overheal": undefined,
    "sourceID": 4,
    "sourceInstance": 1,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 199865,
    "type": "damage",
    "unmitigatedAmount": 164,
  },
  Object {
    "abilityGameID": 77478,
    "amount": 74383,
    "buffs": "330380.",
    "hitType": 1,
    "overheal": undefined,
    "sourceID": 2,
    "sourceMarker": 2,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 200038,
    "type": "damage",
    "unmitigatedAmount": 1074,
  },
  Object {
    "abilityGameID": 345495,
    "amount": 2916,
    "buffs": "331939.307185.208628.",
    "hitType": 1,
    "overheal": undefined,
    "sourceID": 23,
    "sourceInstance": 8,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 200127,
    "type": "damage",
    "unmitigatedAmount": 463,
  },
  Object {
    "abilityGameID": 133,
    "amount": 153220,
    "buffs": "333030.308488.307185.",
    "hitType": 1,
    "overheal": undefined,
    "sourceID": 3,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 202555,
    "type": "damage",
    "unmitigatedAmount": 1348,
  },
]
`;

exports[`prideful getDamageDoneToManifestationOfPrideEvents (first event only) 1`] = `
Array [
  Object {
    "abilityGameID": 225919,
    "amount": 616,
    "buffs": "337547.",
    "hitType": 1,
    "mitigated": 263,
    "sourceID": 1,
    "sourceMarker": 7,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 196204,
    "type": "damage",
    "unmitigatedAmount": 879,
  },
]
`;

exports[`prideful getDamageTakenByManifestationOfPrideEvents (all events) 1`] = `
Array [
  Object {
    "abilityGameID": 1,
    "absorbed": 899,
    "amount": 120725,
    "buffs": "328900.203981.337547.343013.",
    "hitType": 1,
    "mitigated": 10316,
    "overheal": undefined,
    "sourceID": 27,
    "sourceInstance": 1,
    "targetID": 1,
    "targetMarker": 7,
    "timestamp": 197036,
    "type": "damage",
    "unmitigatedAmount": 19317,
  },
  Object {
    "abilityGameID": 342332,
    "amount": 35915,
    "hitType": 1,
    "mitigated": 307,
    "overheal": undefined,
    "sourceID": 27,
    "sourceInstance": 1,
    "targetID": 6,
    "timestamp": 197805,
    "type": "damage",
    "unmitigatedAmount": 1205,
  },
  Object {
    "abilityGameID": 342332,
    "amount": 43385,
    "hitType": 1,
    "mitigated": 120,
    "overheal": undefined,
    "sourceID": 27,
    "sourceInstance": 1,
    "targetID": 2,
    "targetMarker": 2,
    "timestamp": 197805,
    "type": "damage",
    "unmitigatedAmount": 1205,
  },
  Object {
    "abilityGameID": 342332,
    "amount": 45252,
    "buffs": "24858.",
    "hitType": 1,
    "mitigated": 74,
    "overheal": undefined,
    "sourceID": 27,
    "sourceInstance": 1,
    "targetID": 7,
    "timestamp": 197805,
    "type": "damage",
    "unmitigatedAmount": 1205,
  },
  Object {
    "abilityGameID": 342332,
    "absorbed": 1118,
    "amount": 43618,
    "buffs": "235313.",
    "hitType": 3,
    "mitigated": 87,
    "overheal": undefined,
    "sourceID": 27,
    "sourceInstance": 1,
    "targetID": 3,
    "timestamp": 197805,
    "type": "damage",
    "unmitigatedAmount": 1205,
  },
]
`;

exports[`prideful getDamageTakenByManifestationOfPrideEvents (first event only) 1`] = `
Array [
  Object {
    "abilityGameID": 1,
    "absorbed": 899,
    "amount": 8102,
    "buffs": "328900.203981.337547.343013.",
    "hitType": 1,
    "mitigated": 10316,
    "sourceID": 27,
    "sourceInstance": 1,
    "targetID": 1,
    "targetMarker": 7,
    "timestamp": 197036,
    "type": "damage",
    "unmitigatedAmount": 19317,
  },
]
`;

exports[`prideful getManifestationOfPrideDeathEvents 1`] = `
Array [
  Object {
    "abilityGameID": 0,
    "sourceID": -1,
    "targetID": 27,
    "targetInstance": 1,
    "timestamp": 224099,
    "type": "death",
  },
  Object {
    "abilityGameID": 0,
    "sourceID": -1,
    "targetID": 27,
    "targetInstance": 2,
    "timestamp": 417930,
    "type": "death",
  },
  Object {
    "abilityGameID": 0,
    "sourceID": -1,
    "targetID": 27,
    "targetInstance": 3,
    "timestamp": 584863,
    "type": "death",
  },
  Object {
    "abilityGameID": 0,
    "sourceID": -1,
    "targetID": 27,
    "targetInstance": 4,
    "timestamp": 921752,
    "type": "death",
  },
  Object {
    "abilityGameID": 0,
    "sourceID": -1,
    "targetID": 27,
    "targetInstance": 5,
    "timestamp": 1390754,
    "type": "death",
  },
]
`;

exports[`prideful getManifestationOfPrideSourceID 1`] = `33`;
